src/NuGet.Clients/NuGet.CommandLine/Common/ConsoleCredentialProvider.cs

```diff
@@ -13,7 +13,7 @@ public class ConsoleCredentialProvider : Credentials.ICredentialProvider
         public ConsoleCredentialProvider(IConsole console)
         {
             Console = console;
-            Id = $"{typeof(ConsoleCredentialProvider).Name}_{Guid.NewGuid()}";
+            Id = $"{nameof(ConsoleCredentialProvider)}_{Guid.NewGuid()}";
```

Comment:
Is there a rule we can enable for this?

Reply:
there is, but only for Rider and R#. the editorconfig key is `resharper_use_name_of_instead_of_type_of_highlighting`. want me to add it? it will light up for r# and rider, and be ignored by everything else

Reply:
I think it's fine to leave it out in that case. 
We'd basically want it to be raised at build time.